### PR TITLE
doc: dev_process: Add description about GH labels usage

### PR DESCRIPTION
**Pull request by aescolar**
There has been some confusion about how GitHub labels should be used.
This commit adds a new doc page, which tries to clarify this.
This discussion started in this doc:
https://docs.google.com/document/d/1EnWOJERsdygOMnymlhKbXkuA5mGPhd9OyX9htJBKXMk/edit

UPDATE: convenience bookmark: https://docs.zephyrproject.org/latest/development_process/dev_env_and_tools.html#labeling-issues-and-pull-requests-in-github
Follow-up in issue #15156 

--------------

Once this PR is merged I will update the labels descriptions in GH to align them with the ones in this page.
Moreover I will delete the following labels (unless somebody disagrees):
```
CII: Silver
won’t fix   (Default GH label) instead describe in a comment why, and close the issue.
invalid               	“                          	”
task
todo
Revisit     Seems inconsistent, better to use milestone tags to indicate when to revisit (?)
TEE         Duplicate of “area: Trusted execution environment” (?)
Process     Duplicate of "area:Process" (?)
```
I will also rename labels to start with uppercase consistently
~~Will rename Trivial => "Trivial PR" (To avoid developers assigning it to trivial bugs.~~

### CODE REVIEW

File: doc/development_process/gh_labels.rst
Comment (dbkinder) [inline doc/development_process/gh_labels.rst]:
```suggestion
An area maintainer should be able to filter by an area label and
```

File: doc/development_process/gh_labels.rst
Comment (dbkinder) [inline doc/development_process/gh_labels.rst]:
```suggestion
Description    To classify the impact and importance of a bug or feature
```

File: doc/development_process/gh_labels.rst
Comment (dbkinder) [inline doc/development_process/gh_labels.rst]:
```suggestion
Note: Issue priorities are generally set or changed during the issue triage or TSC.
```

File: doc/development_process/gh_labels.rst
Comment (dbkinder) [inline doc/development_process/gh_labels.rst]: This label is currently called ``waiting for response``.  This name should also be enclosed in backticks to be formatted like all the others in this table.

File: doc/development_process/gh_labels.rst
Comment (dbkinder) [inline doc/development_process/gh_labels.rst]: Is this an opportunity to have all labels with a consistent upper-case first letter?

File: doc/development_process/gh_labels.rst
Comment (dbkinder) [inline doc/development_process/gh_labels.rst]:
```suggestion
``Regression``       Something, which was working, but does not anymore
```

File: doc/development_process/gh_labels.rst
Comment (dbkinder) [inline doc/development_process/gh_labels.rst]: A consistent first upper-case letter would be good...

Also, are we encouraging folks to ask questions as GitHub issues, or asking them to post questions in Slack?

File: doc/development_process/gh_labels.rst
Comment (dbkinder) [inline doc/development_process/gh_labels.rst]: Maybe add:
```
Issues and PRs often have multiple labels assigned,
to help classify them in the different available categories.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/development_process/gh_labels.rst
Comment (aescolar) [reply doc/development_process/gh_labels.rst]: I can certainly fix that :)

File: doc/development_process/gh_labels.rst
Comment (aescolar) [reply doc/development_process/gh_labels.rst]: >A consistent first upper-case letter would be good..

Sure

> Also, are we encouraging folks to ask questions as GitHub issues, or asking them to post questions in Slack?

Well that is another thing. The fact is that today, people comes to both places with questions, and some issues are morphed into questions.

File: doc/development_process/gh_labels.rst
Comment (aescolar) [reply doc/development_process/gh_labels.rst]: I take the 1st change. About the 2nd, the meeting invite actually calls it "Bug Triage & Release Readiness", so I was trying to be consistent with that.

**Comment by aescolar** [comment]
@dbkinder Like always, thanks David. Your comments should be addressed now (except for what was pointed above) Please take another look and tell if you find anything else.

File: doc/development_process/gh_labels.rst
Comment (dbkinder) [inline doc/development_process/gh_labels.rst]: I thought I had added a comment about this, but doesn't look like I saved it.  Instead of using the word "area" to define what an "area" is, how about something like:
```
Indicates subsystems (e.g., Kernel, I2C, Memory Management),
project functions (e.g., Debugging, Documentation, Process), or other
categories (e.g., Coding Style, MISRA-C)  affected by the bug or pull request.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
Looking better.  I left off a comment on the previous review...

File: doc/development_process/gh_labels.rst
Comment (aescolar) [reply doc/development_process/gh_labels.rst]: Done :)

**Review by dbkinder — APPROVED** [review]
LGTM, thanks!

File: doc/development_process/gh_labels.rst
Comment (carlescufi) [inline doc/development_process/gh_labels.rst]: Can you link to the GitHub Wiki pages where those meetings are described?

File: doc/development_process/gh_labels.rst
Comment (carlescufi) [inline doc/development_process/gh_labels.rst]: isn't the labe "Trivial" and not "Trivial PR"?

File: doc/development_process/gh_labels.rst
Comment (carlescufi) [inline doc/development_process/gh_labels.rst]: Is there somewhere with more information about what those categories mean? For example in the development process we should have how many of those we area allowed to have open before release

File: doc/development_process/gh_labels.rst
Comment (carlescufi) [inline doc/development_process/gh_labels.rst]: I think it's important to note that an issue needs to have one of the 4 following labels:

- Bug
- Question
- Enhancement
- Feature Request

**Review by carlescufi — CHANGES_REQUESTED** [review]
Nice work @aescolar thanks for this! I've added a few comments

File: doc/development_process/gh_labels.rst
Comment (aescolar) [reply doc/development_process/gh_labels.rst]: The plan was to rename Trivial => "Trivial PR" (To avoid developers assigning it to trivial bugs.

File: doc/development_process/gh_labels.rst
Comment (aescolar) [reply doc/development_process/gh_labels.rst]: Not that I'm aware. But yes, I agree it is important to clarify it. As soon as we would, it should be linked from here.

Comment (carlescufi) [reply doc/development_process/gh_labels.rst]: I thought we did, strange. @nashif ?

File: doc/development_process/gh_labels.rst
Comment (carlescufi) [reply doc/development_process/gh_labels.rst]: Does it matter if we reuse the label for bugs and PRs?

Comment (aescolar) [reply doc/development_process/gh_labels.rst]: If there is any purpose of using a "Trivial" label in a bug then we could use it. Today, I don't see any need for such.
But having too many labels with no clear purpose will add to the confusion (which we are trying to reduce with this PR)

File: doc/development_process/gh_labels.rst
Comment (mbolivar) [inline doc/development_process/gh_labels.rst]: nit: this colon (":") seems like it should be removed

File: doc/development_process/gh_labels.rst
Comment (mbolivar) [inline doc/development_process/gh_labels.rst]: I'm having a little trouble understanding what "reduces the changes of duplicates" means -- do you think you can clarify this?

File: doc/development_process/gh_labels.rst
Comment (mbolivar) [inline doc/development_process/gh_labels.rst]: Should we consider deprecating this label in favor of using draft pull requests only? https://github.blog/2019-02-14-introducing-draft-pull-requests/

I see below that `This PR should not be merged (Do Not Merge).  For work in progress, GitHub "draft" PRs are preferred"` is listed next to DNM. I also see `In progress` is a label which can apply to PRs that seems very similar.

As a reader, I'm a bit confused between the following labels for pull requests, which all seem to be related:

 - RFC
- In progress
- DNM
- GitHub "Draft" PRs

Is there an opportunity to consolidate these into just one label (or perhaps just use draft PRs)?

**Review by mbolivar — COMMENTED** [review]
Looks nice, thanks a lot for doing this!

I have a minor grammar nit and then some questions about the labels for "incomplete" or non-mergeable PRs.

File: doc/development_process/gh_labels.rst
Comment (aescolar) [reply doc/development_process/gh_labels.rst]: would it be enough to say "reduces the changes of duplicate issue reports"?
(it is just a quick motivational paragraph, I would not it want to become too long)

File: doc/development_process/gh_labels.rst
Comment (aescolar) [reply doc/development_process/gh_labels.rst]: Maybe, but not trivially right now.
(Note that I'm not trying to fix all issues related to our way of working with this PR)
I'm hoping to start by document what we have, and at the same time, clean what is trivial to clean.
If we start by going further in this same PR it will just balloon out of proportion.
=> If you don't mind, let's take that discussion separately from this PR.

File: doc/development_process/gh_labels.rst
Comment (aescolar) [reply doc/development_process/gh_labels.rst]: would a semicolon seem better?

Comment (mbolivar) [reply doc/development_process/gh_labels.rst]: I don't think so. A semicolon would mean "to classify them by area, type, [...] attention" is an independent clause, which it is not (https://www.proofreadnow.com/blog/bid/71799/The-Semicolon-and-Independent-Clauses).

I don't think any punctuation marks are needed (or appropriate)

File: doc/development_process/gh_labels.rst
Comment (mbolivar) [reply doc/development_process/gh_labels.rst]: OK, so this is about avoiding dupes? How about "avoids duplicate issue reports"?

File: doc/development_process/gh_labels.rst
Comment (mbolivar) [reply doc/development_process/gh_labels.rst]: Sure

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/development_process/gh_labels.rst
Comment (aescolar) [reply doc/development_process/gh_labels.rst]: Thanks, done

File: doc/development_process/gh_labels.rst
Comment (aescolar) [inline doc/development_process/gh_labels.rst]: @carlescufi: Should we separate the API label into a "area: API", and "API (meeting)", to clear a bit things?

Comment (dbkinder) [inline doc/development_process/gh_labels.rst]: There's an organized person in me that would like to have all labels with a prefix (e.g., ``area: API``, ``meetings: API``) so that the labels are grouped together (``meetings: API``, ``meetings: Dev-review``, ``meetings: TSC``), otherwise they don't sort together.

File: doc/development_process/gh_labels.rst
Comment (dbkinder) [inline doc/development_process/gh_labels.rst]: Is this the same ``TSC`` mentioned above?  If it's different, then a prefix would clarify :)

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/development_process/gh_labels.rst
Comment (aescolar) [reply doc/development_process/gh_labels.rst]: I agree :)

File: doc/development_process/gh_labels.rst
Comment (aescolar) [reply doc/development_process/gh_labels.rst]: Today it is the same, I would be happy to split it in two, but Anas was not too keen on it.

File: doc/development_process/gh_labels.rst
Comment (aescolar) [reply doc/development_process/gh_labels.rst]: Would it be reasonable to, by now, have this PR start as documentation of what there actually is, and then open an issue with this different fixes? (and have them applied once agreed in a a following PR). I just don't want this PR to drag long.

Comment (dbkinder) [reply doc/development_process/gh_labels.rst]: OK.  Maybe you can create an issue to track this (adding a prefix to all labels).

**Review by dbkinder — APPROVED** [review]
we can merge this now to document how labels are used, and submit a future fix for resolving label organization and specific names...

File: doc/development_process/review_process.rst
Comment (nashif) [inline doc/development_process/review_process.rst]: Please keep it as Trivial, no need for the PR addition.

File: doc/development_process/review_process.rst
Comment (nashif) [inline doc/development_process/review_process.rst]: This is already indicated in the top section, it is about time, keep it general, TSC and Security and labels are not related to time.

File: doc/development_process/review_process.rst
Comment (nashif) [inline doc/development_process/review_process.rst]: and linked

File: doc/development_process/gh_labels.rst
Comment (nashif) [inline doc/development_process/gh_labels.rst]: can this document be integrated into the existing issues document as a sub-section?

File: doc/development_process/gh_labels.rst
Comment (nashif) [reply doc/development_process/gh_labels.rst]: dont like the meeting prefix, it does not have to be meeting, this is something that can always be brought into discussion, on slack, in email and other places.

File: doc/development_process/gh_labels.rst
Comment (nashif) [reply doc/development_process/gh_labels.rst]: Not sure this is documented anywhere beside TSC meeting minutes. Need to add.

File: doc/development_process/gh_labels.rst
Comment (nashif) [inline doc/development_process/gh_labels.rst]: And issue can block another issue, so this applies to more than just PRs

**Review by nashif — CHANGES_REQUESTED** [review]

File: doc/development_process/gh_labels.rst
Comment (dbkinder) [reply doc/development_process/gh_labels.rst]: maybe ``review:`` then?

File: doc/development_process/review_process.rst
Comment (aescolar) [reply doc/development_process/review_process.rst]: Ok, I will revert. The reason for the change was due to the headings' levels not being to clear, and somebody maybe understanding this to be the general category/label section.

File: doc/development_process/review_process.rst
Comment (aescolar) [reply doc/development_process/review_process.rst]: thanks, indeed.

File: doc/development_process/gh_labels.rst
Comment (aescolar) [reply doc/development_process/gh_labels.rst]: Today that mini-page content is a bit disconnected from this one (I guess you meant issues.rst ?)
You have a PR editing that one.. So after this is merged, feel free to take this content in if you feel it fits better there.

File: doc/development_process/gh_labels.rst
Comment (aescolar) [reply doc/development_process/gh_labels.rst]: correct, moved

File: doc/development_process/gh_labels.rst
Comment (marc-hb) [reply doc/development_process/gh_labels.rst]: The only way to make that clear and simple is for them to share a common prefix, I suggest: `Type: bug`, `Type: Question`,... This will also group them in the list.

File: doc/development_process/gh_labels.rst
Comment (carlescufi) [reply doc/development_process/gh_labels.rst]: I don't think it's necessary. API is API, we can go through everything marked as API during the meeting

**Review by carlescufi — APPROVED** [review]

File: doc/development_process/gh_labels.rst
Comment (aescolar) [reply doc/development_process/gh_labels.rst]: My thinking was that such a label can both help you to filter, but also the author of the PR, to know the PR is going to be discussed there.

File: doc/development_process/index.rst
Comment (nashif) [inline doc/development_process/index.rst]: IMO this should not be a top level section, it should go under something else like issues or proposals or the dev process.

**Review by nashif — CHANGES_REQUESTED** [review]

**Comment by aescolar** [comment]
Compared to what people had reviewed:
Following [this request](https://github.com/zephyrproject-rtos/zephyr/pull/15054#pullrequestreview-224923911) from @nashif the labels section is now moved into the "Developemnt Environment and Tools" page.
Due to this, all the headings are now 1 level lower to align them with that page.

Apart from that the only other change, due to a new reference added in 15ca53dbbab, the equivalent new reference which had been added in this PR is now removed.
```diff
diff --git a/doc/development_process/dev_env_and_tools.rst b/doc/development_process/dev_env_and_tools.rst
index 43a626054f..c46965e0af 100644
--- a/doc/development_process/dev_env_and_tools.rst
+++ b/doc/development_process/dev_env_and_tools.rst
@@ -15,7 +15,7 @@ patches, which are then reviewed by the project members before being applied to
 the code base.
 
 Pull requests should be appropriately :ref:`labeled<gh_labels>`,
-and linked to any relevant :ref:`bug or feature tracking issues<issue_tracking>`
+and linked to any relevant :ref:`bug or feature tracking issues<bug_reporting>`
 .
 
 The Zephyr project uses GitHub for code reviews and Git tree management. When
diff --git a/doc/development_process/issues.rst b/doc/development_process/issues.rst
index 16552f1a49..ceca4d095b 100644
--- a/doc/development_process/issues.rst
+++ b/doc/development_process/issues.rst
@@ -1,5 +1,3 @@
-.. _issue_tracking:
-
 .. _bug_reporting:
 
 Bug Reporting
```

**Review by nashif — APPROVED** [review]

**Pull request merged**
